examples/rotation_bin/rotation_bin/imp.rs

```diff
@@ -0,0 +1,247 @@
+        // Otherwise return child's measure but discard minimum_baseline and natural_baseline
+        // as baseline is only useful for vertical orientations
+        match (orientation, self.rotation(widget)) {
+            (gtk::Orientation::Horizontal, Rotation::Normal | Rotation::Deg180) => {
+                child.measure(orientation, for_size)
+            }
+            (_, _) => {
+                let (min, nat, _, _) = child.measure(orientation, for_size);
+                (min, nat, -1, -1)
+            }
+        }
+    }
+
+    fn size_allocate(&self, widget: &Self::Type, width: i32, height: i32, baseline: i32) {
+        let child = self.child.borrow();
+        let child = match child.as_ref() {
+            Some(child) => child,
+            None => return,
+        };
+
+        // Swap height and width for 90 or 270 degrees
+        let (w, h) = match self.rotation(widget) {
+            Rotation::Deg90 | Rotation::Deg270 => (height, width),
+            _ => (width, height),
+        };
+
+        let transform = gsk::Transform::new();
+        // Transformation are done with respect to their origin point, top-left by default in GTK.
+        // Move the origin of the transformation to the center, instead of a corner, by taking half
+        // of width and height.
+        let transform = transform
+            .translate(&Point::new(width as f32 / 2.0, height as f32 / 2.0))
+            .unwrap();
+        // Rotate by the desired angle
+        let transform = transform.rotate(self.rotation(widget).into()).unwrap();
+        // Revert the move of the origin point once our rotation is done.
+        let transform = transform.translate(&Point::new(-w as f32 / 2.0, -h as f32 / 2.0));
+        child.allocate(w, h, baseline, transform.as_ref())
+    }
+}
+
+impl RotationBin {
+    pub(super) fn child(&self, _obj: &super::RotationBin) -> Option<gtk::Widget> {
+        self.child.borrow().clone()
+    }
+
+    pub(super) fn rotation(&self, _obj: &super::RotationBin) -> Rotation {
+        self.rotation.get()
+    }
+
+    pub(super) fn set_child(
+        &self,
+        obj: &super::RotationBin,
+        widget: Option<&impl IsA<gtk::Widget>>,
+    ) {
+        let widget = widget.map(|w| w.upcast_ref());
+        if widget == self.child.borrow().as_ref() {
+            return;
+        }
+
+        if let Some(child) = self.child.borrow_mut().take() {
+            child.unparent();
+        }
+
+        if let Some(w) = widget {
+            self.child.replace(Some(w.clone()));
+            w.set_parent(obj);
+        }
+
+        obj.queue_resize();
+        obj.notify("child")
+    }
+
+    pub(super) fn set_rotation(&self, obj: &super::RotationBin, rotation: Rotation) {
+        if self.rotation.get() != rotation {
+            self.rotation.replace(rotation);
+            obj.queue_resize();
+            obj.notify("rotation");
+        }
+    }
+
+    pub(super) fn rotate_clockwise(&self, obj: &super::RotationBin) {
```

Comment:
You don't need to pass super::RotationBin here, you can get it from the private struct with `self.instance()` https://gtk-rs.org/gtk-rs-core/git/docs/glib/subclass/types/trait.ObjectSubclassExt.html#tymethod.instance

Reply:
Yes, but I kinda like it having an obj param like the rest of the gtk/gst methods do

Reply:
They do so because they are virtual methods and are there for convenience. I do wonder though why we are not using self.instance there, cc @sdroege

Reply:
Because you often need it and it's annoying to have to call another function every time :)